### PR TITLE
[ADD] main_branch_bot: _gen_addons_icon

### PR DESCRIPTION
**Pull request by mreficent**
Depends on:

- [x] https://github.com/OCA/maintainer-tools/pull/411.

### CODE REVIEW

**Comment by sbidoul** [comment]
@mreficent thanks! You need to format the code with black to get travis green.
The easiest is with pre-commit as explained in the [development section of the readme](https://github.com/OCA/oca-github-bot#development).

**Comment by mreficent** [comment]
travis-ci obviously is not green because the other PR is not merged yet.

**Comment by sbidoul** [comment]
I'm afraid this part is not covered by tests yet.... The reason travis is red is code style.

**Comment by mreficent** [comment]
`black` requires Python 3.6.0+ and I am using Python 3.5.2 and I don't want to install an upper version :S

**Comment by sbidoul** [comment]
Are you on Ubuntu? Try the [deadsnakes ppa](https://launchpad.net/~deadsnakes/+archive/ubuntu/ppa). It's painless.

**Comment by mreficent** [comment]
@sbidoul Already green :green_apple: , ready to merge? :)

**Comment by sbidoul** [comment]
@mreficent I can merge. Are ou extra sure it will do no harm? When I deploy this it will run on all OCA repos.

**Comment by mreficent** [comment]
I am sure it will do no harm, but you know, the unexpected is always unexpected. The test is in https://github.com/OCA/maintainer-tools, and works fine, and the method is simple: checks if it's a module or not, and also only creates the `.png` if no `.png` or `.svg` is found in `static/description/`. It doesn't do anything more or anything less.

**Comment by sbidoul** [comment]
Thanks. I'll deploy soon.

**Comment by sbidoul** [comment]
And it's deployed.

**Comment by sbidoul** [comment]
@mreficent it failed with `"fatal: pathspec './event_phone/static/description/icon.png'` in OCA/connector-telephony 11. Not sure why. I made this https://github.com/OCA/maintainer-tools/pull/416, but that does not seem enough to explain the error.

Sentry did not give a full stack trace, but that errors seems to come from git.

**Comment by mreficent** [comment]
But your PR solves the issue? :S

No more info of the issue?

**Comment by sbidoul** [comment]
I'll deploy the PR but I don't think it solves the issue. 

The error in Sentry looks like this:
```b'fatal: pathspec \'./project_task_stage_allow_timesheet/static/description/icon.png\' did not match any files'```

File: Dockerfile [hunk 1]
```diff
@@ -24,7 +24,8 @@ RUN set -x \
 RUN set -x \
   && /ocamt/bin/pip install git+https://github.com/OCA/maintainer-tools@d9e2c3b8dc163c8d0fcdd36278b782074c3cff12 \
```
Comment (mreficent) [inline Dockerfile]: shouldn't this hash be changed? 🤔

Comment (mreficent) [reply Dockerfile]: Ah, nothing, I see you updated that hahaha

**Comment by sbidoul** [comment]
@mreficent I think I got it: bcf7b4806f8bfa423743002a3e00ba9ca9d4319d da9ef6f187464e377519319aee1b63576bef68e4  I redeployed.

**Comment by mreficent** [comment]
Good! Genius! :+1: :1st_place_medal:

**Comment by sbidoul** [comment]
It works fine now: https://github.com/OCA/oca-custom/commits/11.0

Thanks for the contrib!

**Pull request merged**